packages/trace-viewer/src/ui/errorsTab.tsx

```diff
@@ -21,6 +21,51 @@ import { PlaceholderPanel } from './placeholderPanel';
 import { renderAction } from './actionList';
 import type { Language } from '@isomorphic/locatorGenerators';
 import type { StackFrame } from '@protocol/channels';
+import { CopyToClipboardTextButton } from './copyToClipboard';
+import { attachmentURL } from './attachmentsTab';
+import { fixTestPrompt } from '@web/components/prompts';
+import type { GitCommitInfo } from '@testIsomorphic/types';
+
+export const GitCommitInfoContext = React.createContext<GitCommitInfo | undefined>(undefined);
+
+const PromptButton: React.FC<{
+  error: string;
+  actions: modelUtil.ActionTraceEventInContext[];
+}> = ({ error, actions }) => {
+  const [pageSnapshot, setPageSnapshot] = React.useState<string>();
+
+  React.useEffect(( )=> {
+    for (const action of actions) {
+      for (const attachment of action.attachments ?? []) {
+        if (attachment.name === 'pageSnapshot') {
+          fetch(attachmentURL({ ...attachment, traceUrl: action.context.traceUrl })).then(async response => {
+            setPageSnapshot(await response.text());
+          });
+          return;
+        }
+      }
+    }
+  }, [actions]);
+
+  const gitCommitInfo = React.useContext(GitCommitInfoContext);
```

Comment:
If you're going to go to the trouble of creating a context, I would recommend that you provide a nice hook and provider variable.